indico/modules/rb_new/client/js/common/bookings/BookingEdit.jsx

```diff
@@ -0,0 +1,622 @@
+                                <Singular>
+                                    Your new booking might conflict with another one.
+                                </Singular>
+                                <Plural>
+                                    <Param name="count" value={this.numberOfConflicts} /> occurrences of your
+                                    new booking might conflict with existing bookings.
+                                </Plural>
+                            </PluralTranslate>
+                        </Message.Content>
+                    </Message>
+                    {this.numberOfCandidates > this.numberOfConflicts && (
+                        <Segment attached="bottom">
+                            <Checkbox toggle
+                                      defaultChecked={skipConflicts}
+                                      label={Translate.string('I understand, please skip any days with conflicting ' +
+                                                              'occurrences.')}
+                                      onChange={(__, {checked}) => {
+                                          this.setState({
+                                              skipConflicts: checked
+                                          });
+                                      }} />
+                        </Segment>
+                    )}
+                </div>
+            );
+        }
+
+        return null;
+    };
+
+    updateBooking = async (data) => {
+        const {dates: {startDate, endDate}, timeSlot: {startTime, endTime}, usage, user, reason, recurrence} = data;
+        const {actions: {updateBooking}, booking: {id, room: {id: roomId}}, onSubmit} = this.props;
+        const {0: repeatFrequency, 1: repeatInterval} = serializeRecurrenceInfo(recurrence);
+        const params = {
+            start_dt: `${startDate} ${startTime}`,
+            end_dt: `${endDate ? endDate : startDate} ${endTime}`,
+            repeat_frequency: repeatFrequency,
+            repeat_interval: repeatInterval,
+            room_id: roomId,
+            user_id: usage === 'myself' ? undefined : user.id,
+            booking_reason: reason,
+        };
+
+        const rv = await updateBooking(id, params);
+        if (rv.error) {
+            return rv.error;
+        } else {
+            onSubmit();
+        }
+    };
+
+    render() {
+        return (
+            <FinalForm onSubmit={this.updateBooking}
+                       render={this.renderBookingEditModal}
+                       initialValues={this.formInitialValues}
+                       subscription={{
+                           submitting: true,
+                           submitSucceeded: true,
+                           hasValidationErrors: true,
+                           pristine: true,
+                           values: true,
+                       }}
+                       keepDirtyOnReinitialize />
```

Comment:
`initialValuesEqual={_.isEqual}` is probably the better option here; you don't want the form to reinitialize so your initialValues shouldn't actually change